tasks/mssql_conf_setting.yml

```diff
@@ -0,0 +1,36 @@
+- name: Check if the {{ __mssql_conf_setting }} setting is enabled
+  command: >-
+    /opt/mssql/bin/mssql-conf get {{ __mssql_conf_setting }}
+  changed_when: false
+  register: __mssql_conf_get_setting
+
+- name: Configure the {{ __mssql_conf_setting }} setting
+  command: >-
+    /opt/mssql/bin/mssql-conf set {{ __mssql_conf_setting }}
+    {{ __mssql_conf_setting_value | quote }}
+  vars:
+    __mssql_conf_get_value: >-
+      {{ __mssql_conf_get_setting.stdout |
+      regex_replace('^.*\s:\s', '') }}
+  when: >-
+    (__mssql_conf_setting_value != 'unset') and
+    (("No setting for the given" in __mssql_conf_get_setting.stdout) or
+    ((__mssql_conf_setting_value | type_debug != "bool") and
+    (__mssql_conf_setting_value | string | lower not in
+    __mssql_conf_get_value | lower)) or
+    ((__mssql_conf_setting_value | type_debug == "bool") and
+    (__mssql_conf_setting_value != __mssql_conf_get_value | bool)))
+  register: __mssql_conf_set
+  failed_when: >
+    ("error" in __mssql_conf_set.stdout | lower) or (__mssql_conf_set.rc == 1)
+  notify: Restart the mssql-server service
+
+- name: Unset the {{ __mssql_conf_setting }} setting
+  command: /opt/mssql/bin/mssql-conf unset {{ __mssql_conf_setting }}
+  when:
+    - '__mssql_conf_setting_value == "unset"'
+    - '"No setting for the given" not in __mssql_conf_get_setting.stdout'
```

Comment:
You do need the single quotes here since the first argument is a quoted string value

Reply:
I removed the single quotes here and it caused and error:
```
fatal: [127.0.0.3]: FAILED! => {"reason": "We were unable to read either as JSON nor YAML, these are the errors we got from each:\nJSON: Expecting value: line 1 column 1 (char 0)\n\nSyntax Error while loading YAML.\n  did not find expected '-' indicator\n\nThe error appears to be in '/home/spetrosi/Documents/work/linux-system-roles/mssql/tasks/mssql_conf_setting.yml': line 32, column 34, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n    - __mssql_conf_setting_value == \"unset\"\n    - \"No setting for the given\" not in __mssql_conf_get_setting.stdout\n                                 ^ here\n"}
```

Reply:
Right - you need the single quotes on the second condition because it starts with a double quoted string